app/controllers/admin/users_controller.rb

```diff
@@ -40,6 +40,30 @@ def update
       redirect_to "/admin/users/#{params[:id]}"
     end
 
+    def destroy
+      role = params[:role].to_sym
+      resource_type = params[:resource_type]
+
+      if role == :super_admin
+        flash[:danger] = "Super Admin roles cannot be removed."
+        redirect_to "/admin/users/#{params[:id]}/edit" and return
+      end
+
+      @user = User.find(params[:user_id])
+
+      if @user.id == current_user.id
+        flash[:danger] = "Admins cannot remove roles from themselves."
+      elsif role == :single_resource_admin
+        User.find(params[:user_id]).remove_role(role, resource_type.safe_constantize)
```

Comment:
**problem (of sorts), blocking:** One thing to think about: `safe_constantize` potentially returns `nil` and `remove_role(:single_resource_admin, nil)` doesn't error will work but may not do what we want it to:

```ruby
user.roles.pluck(:name, :resource_type)
#=> [["super_admin", nil], ["single_resource_admin", "Config"]]
resouce_type = "oops"
user.remove_role(:single_resource_admin, resource_type.safe_constantize)
user.roles.pluck(:name, :resource_type)
#=> [["super_admin", nil]]
```

As you can see the `:single_resource_admin` role for `Config` is gone, even though the `resource_type` was non-existing. So we should probably add a check to make sure the result of `safe_constantize` isn't `nil`. I know that because of the way the form is built this shouldn't happen, but I think I'd still feel better if we address this.

Reply:
Done! ✔️